core/src/main/java/jenkins/telemetry/Telemetry.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.telemetry;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.ProxyConfiguration;
+import hudson.model.AsyncPeriodicWork;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import hudson.model.TaskListener;
+import hudson.model.UsageStatistics;
+import jenkins.model.Jenkins;
+import jenkins.util.SystemProperties;
+import net.sf.json.JSONObject;
+
+import javax.annotation.Nonnull;
+import javax.net.ssl.HttpsURLConnection;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Extension point for collecting JEP-214 telemetry.
+ *
+ * @see <a href="https://github.com/jenkinsci/jep/tree/master/jep/214">JEP-214</a>
+ */
+public abstract class Telemetry implements ExtensionPoint {
+    /**
+     * ID of this collector, typically a basic alphanumeric string (and _- characters)
+     * 
+     * @return ID of the collector, never null or empty
+     */
+    @Nonnull
+    public abstract String getId();
```

Comment:
Yes, that's the idea (although I suspect in practice it might be good enough to be unique during the collection period).

If this is a major concern I can remove this entirely, and we'll just use the class name (including package) instead? That should be reasonably unique…?